pymc_marketing/clv/models/beta_geo.py

```diff
@@ -437,6 +439,71 @@ def expected_probability_alive(
             "chain", "draw", "customer_id", missing_dims="ignore"
         )
 
+    def expected_probability_no_purchase(
+        self,
+        t: int,
+        data: pd.DataFrame | None = None,
+    ) -> xarray.DataArray:
+        r"""Compute the probability a customer with history frequency, recency, and T
+        will have 0 purchases in the period (T, T+t].
+
+        The data parameter is only required for out-of-sample customers.
+
+        Adapted from Section 5.3, Equation 34 in Bruce Hardie's notes [1]_ for the special case when y = 0
+
```

Comment:
Can we change _y_ to _frequency_ here for clarify?

Reply:
I have removed that part as suggested. This special case is indeed not mentioned in the research notes of Hardie. It is simply a special case I had derived. The derivation of this is linked in the PDF attached in the [original issue](https://github.com/pymc-labs/pymc-marketing/issues/1093) of this PR.